### PR TITLE
Test::Quattor::command_history_ok: add 2nd nocommand argument

### PR DESCRIPTION
**Pull request by stdweird**

### CODE REVIEW

File: build-scripts/src/main/perl/Test/Quattor.pm [hunk 1]
```diff
@@ -702,11 +702,38 @@ the fact that x2 was also called but not checked is allowed.).
 C<command_history_ok([x3,x2])> returns 0 (wrong order),
 C<command_history_ok([x1,x4])> returns 0 (no x4 command).
 
+A second arrayref of command(regexp)s can be given,
+and the C<@command_history> is then first checked that none of those
+occured.
+If you only want to check with the non-occurence of commands,
```
Comment (jouvin) [inline build-scripts/src/main/perl/Test/Quattor.pm]: `check` rather than `check with`?

File: build-scripts/src/main/perl/Test/Quattor.pm [hunk 2]
```diff
@@ -702,11 +702,38 @@ the fact that x2 was also called but not checked is allowed.).
...
+pass an undef as the first argument
```
Comment (jouvin) [inline build-scripts/src/main/perl/Test/Quattor.pm]: I am not completely sure to understand the comment. Which commands are checked against the forbidden commands when command history is `undef`?

File: build-scripts/src/main/perl/Test/Quattor.pm [hunk 3]
```diff
@@ -702,11 +702,38 @@ the fact that x2 was also called but not checked is allowed.).
...
 
 sub command_history_ok
 {
-    my $commands = shift;
+    my ($commands, $nocommands) = @_;
+
+    if ($nocommands) {
+        foreach my $cmd (@$nocommands) {
+            if (grep {$_ =~ /$cmd/} @command_history) {
+                diag "command_history_ok nocommand found '$cmd' in history; return false";
```
Comment (jouvin) [inline build-scripts/src/main/perl/Test/Quattor.pm]: I found this message not very clear... What about `command_history_ok: '$cmd' found in the list of forbidden commands; returning false`

File: build-scripts/src/main/perl/Test/Quattor.pm [hunk 4]
```diff
@@ -702,11 +702,38 @@ the fact that x2 was also called but not checked is allowed.).
...
+    }
+
+    if (! defined($commands)) {
+        if ($nocommands) {
+            # This is ok.
+            diag "No commands to test and nocommand check was ok.";
+            return 1;
+        } else {
+            # Fatal
+            ok(0, "command_history_ok: neither commands nor nocommnds specified");
```
Comment (jouvin) [inline build-scripts/src/main/perl/Test/Quattor.pm]: `nocommds` -> `nocommands`.

In the message, 'neither commands nor nocommands **argument**'

File: build-scripts/src/main/perl/Test/Quattor.pm [hunk 5]
```diff
@@ -702,11 +702,38 @@ the fact that x2 was also called but not checked is allowed.).
...
+If you only want to check with the non-occurence of commands,
+pass an undef as the first argument
+(and not an empty arrayref).
+
 =cut
 
 sub command_history_ok
 {
-    my $commands = shift;
+    my ($commands, $nocommands) = @_;
```
Comment (jouvin) [inline build-scripts/src/main/perl/Test/Quattor.pm]: `nocommands` -> `forbidden_cmds`?

**Comment by stdweird** [comment]
@jouvin thanks for the suggestion of the required vs forbidden command names. i hope it is clearer now

**Comment by jrha** [comment]
LGTM

**Pull request merged**